plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -638,6 +646,11 @@ public Optional<ConnectorOutputMetadata> finishInsert(ConnectorSession session,
                         icebergTable.schema().findType(field.sourceId())))
                 .toArray(Type[]::new);
 
+        if (commitTasks.isEmpty()) {
```

Comment:
Yes, this could be done indeed as you mentioned.
This has been actually the initial version of the code in the PR.

However,  the code would be a bit more difficult to understand with the handling for this special case in `finishInsert` so we've decided to go further with adding the handling for empty CTAS into `finishCreateTable`.

Related discussion:

https://github.com/trinodb/trino/pull/12412#discussion_r873892732

Reply:
Shouldn't this check happen before instantiating ` Type[] partitionColumnTypes` ?

Reply:
I moved the check at the beginning of the method. Thanks for bringing this up.